platform/ext/target/nordic_nrf/Native_Driver/spu.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __SPU_H__
+#define __SPU_H__
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * \brief Reset all memory regions to being Secure
+ *
+ * Reset all (Flash or SRAM) memory regions to being Secure
+ * and have default (i.e. Read-Write-Execute allow) access policy
+ *
+ * \note region lock is not applied to allow modifying the configuration.
+ */
+void spu_regions_reset_all_secure(void);
+
+/**
+ * \brief Configre memory regions as Non-Secure
+ *
+ * Configure a range of (Flash or SRAM) memory regions as Non-Secure
+ *
+ * \note region lock is applied to prevent further modification during
+ * the current reset cycle.
+ */
+
+void spu_regions_config_non_secure( u32_t start_addr, u32_t limit_addr, bool ram);
+
+/**
+ * \brief Configure Non-Secure Callable area
+ *
+ * Configure a single region in Secure Flash as Non-Secure Callable
+ * (NSC) area.
+ *
+ * \note Any Secure Entry functions, exposing secure services to the
+ * Non-Secure firmware, shall be located inside this NSC area.
+ *
+ * If the start address of the NSC area is hard-coded, it must follow
+ * the HW restrictions: The size must be a power of 2 between 32 and
+ * 4096, and the end address must fall on a SPU region boundary.
```

Comment:
Shouldn't it be checked with an assertion within the function?

Reply:
Yes, this is actually done in the downstream SPM - I was just thinking to skip for the first prototype, but I can add the assert (need to find out how TF-M asserts work)

Reply:
EDIT: these asserts are not part of downstream (NCS SPM) run-time checking. So I added the checks as NRF_ASSERT statements. [Checks are done at build time during linking, though]